src/plat/bcm2711/overlay-rpi4.dts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ * Copyright (C) 2021, Hensoldt Cyber GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+/ {
+	chosen {
+		seL4,elfloader-devices =
+		    "serial1";
+
+		seL4,kernel-devices =
+		    "serial1",
+		    &{/soc/interrupt-controller@40041000},
+		    &{/soc/local_intc@40000000},
+		    &{/timer};
+	};
+
+	memory@0 {
+		/* This is configurable in the Pi's config.txt, but we use 128MiB of RAM by default. */
+		reg = <0x00 0x00000000 0x08000000>;
```

Comment:
what is the memory size on the device? why is 128MiB picked?

Reply:
The RPi4 (bcm2711) port is generally based on the RPi3 (bcm2837) port. In the corresponding overlay-rpi3.dts file (https://github.com/seL4/seL4/blob/master/src/plat/bcm2837/overlay-rpi3.dts), also only 128MiB are declared although the RPi3 has more than 128 MiB RAM. Furthermore, the RPi4 comes with different RAM sizes (1,2,4,8 GiB) (https://www.canadarobotix.com/blogs/how-to/how-to-tell-which-raspberry-pi-4-ram-size-do-i-have). I mean, it would be possible to create a different overlay.dts file for the different RPi4 versions. But in the end anyone who wants a different RAM size can modify this file to his needs anyways and the current version seems to work just fine for now.

Reply:
Thanks for the explanation. So the minimal size for all RPi4 boards is 1GiB, which could be used here instead of 128. You mentioned " This is configurable in the Pi's config.txt". What is the config.txt file? Modifying the file can change the usable memory size?

Reply:
The comment also stems from the corresponding RPi3 file. The config.txt file is part of the necessary boot files for Raspis (https://www.raspberrypi.org/documentation/configuration/boot_folder.md). In the config.txt, you can specify configuration parameters for setting up the Raspi. According to https://www.raspberrypi.org/documentation/configuration/config-txt/memory.md, you can specify the gpu_mem size (VideoCore size) and limit the total amount of RAM with total_mem. The minimum value here is 128MiB, which might be an explanation for this configuration.